src/components/widgets/DateRange.vue

```diff
@@ -0,0 +1,88 @@
+<script setup lang="ts">
+import { ref, watch, defineProps, defineEmits } from 'vue';
+import { DateRange } from '@/composables/useSearchParams';
+import EditableDate from '@/components/widgets/EditableDate.vue';
+
+
+const props = defineProps({
+  modelValue: {
+    type: Object,
+    required: true
+  }
+});
```

Comment:
Using v-model is desired as we don't need to keep track of the `emit`

```suggestion
const model = defineModel<{
  start: string,
  end: string
}>({ required: true })
```